src/test/java/teammates/common/datatransfer/questions/FeedbackTextQuestionDetailsTest.java

```diff
@@ -1,14 +1,103 @@
+        assertTrue(feedbackTextQuestionDetails.shouldChangesRequireResponseDeletion(updatedFeedbackTextQuestionDetails));
+
+        ______TS("Updated question does not allow rich text and original question does not allow rich text; return false");
+        feedbackTextQuestionDetails.setShouldAllowRichText(false);
+        updatedFeedbackTextQuestionDetails.setShouldAllowRichText(false);
+        assertFalse(feedbackTextQuestionDetails.shouldChangesRequireResponseDeletion(updatedFeedbackTextQuestionDetails));
+
+        ______TS("Updated question does not allow rich text and original question allows rich text; return false");
+        feedbackTextQuestionDetails.setShouldAllowRichText(false);
+        updatedFeedbackTextQuestionDetails.setShouldAllowRichText(true);
+        assertFalse(feedbackTextQuestionDetails.shouldChangesRequireResponseDeletion(updatedFeedbackTextQuestionDetails));
+    }
+
+    @Test
+    public void testValidateQuestionDetails() {
+        ______TS("Recommended length is null; returns an empty list");
+        feedbackTextQuestionDetails.setRecommendedLength(null);
+        assertTrue(feedbackTextQuestionDetails.validateQuestionDetails().isEmpty());
+
+        ______TS("Recommended length is not null and is less than 1; returns a non-empty list");
+        List<String> expectedResult = new ArrayList<>();
+        expectedResult.add(FeedbackTextQuestionDetails.TEXT_ERROR_INVALID_RECOMMENDED_LENGTH);
+        feedbackTextQuestionDetails.setRecommendedLength(0);
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+
+        feedbackTextQuestionDetails.setRecommendedLength(-1);
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+
+        feedbackTextQuestionDetails.setRecommendedLength(-100);
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+
+        ______TS("Recommended length is not null and is greater than or equal to 1; returns an empty list");
+        expectedResult.clear();
+        feedbackTextQuestionDetails.setRecommendedLength(1);
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+
+        feedbackTextQuestionDetails.setRecommendedLength(2);
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+
+        feedbackTextQuestionDetails.setRecommendedLength(100);
+        assertEquals(expectedResult, feedbackTextQuestionDetails.validateQuestionDetails());
+    }
+
+    @Test
+    public void testValidateResponsesDetails_shouldReturnEmptyList() {
+        List<FeedbackResponseDetails> responses = new ArrayList<>();
+        int numRecipients = 0;
+
+        assertTrue(feedbackTextQuestionDetails.validateResponsesDetails(responses, numRecipients).isEmpty());
+
+        for (int i = 0; i < 10; i++) {
+            responses.add(new FeedbackTextResponseDetails());
+        }
```

Comment:
Perhaps we won't need a loop here, maybe just add once or twice will do? (The method under test is also straightforward so might prefer simpler solutions)

Reply:
Ok, thank you for the advice.